docs/tutorials/ad-hoc-simulation/planning-simulation.md

```diff
@@ -2,81 +2,130 @@
+    Remember to set the initial pose of the car in the same direction as the lane.
 
-   ![delete-objects](images/planning/lane-following/delete-objects.png)
+    To confirm the direction of the lane, check the arrowheads displayed on the map.
 
-7. After that, the vehicle will restart driving and reach the goal.
+#### 4. Set a goal pose for the ego vehicle
 
-   ![restart-driving](images/planning/lane-following/restart-driving.png)
+a) Click the `2D Goal Pose` button in the toolbar, or hit the `G` key.
 
-   ![reach-goal](images/planning/lane-following/reach-goal.png)
+b) In the 3D View pane, click and hold the left-mouse button, and then drag to set the direction for the goal pose. If done correctly, you will see a planned path from initial pose to goal pose.
 
-## How to simulate parking maneuvers
+![set-goal-pose](images/planning/lane-following/set-goal-pose.png)
+
+#### 5. Engage the ego vehicle
+
+Now you can start the ego vehicle driving by clicking the `Engage` button in `AutowareStatePanel`. Alteratively, you can manually engage the vehicle by running the following command:
+
+```bash
+source ~/autoware/install/setup.bash
+ros2 topic pub /autoware/engage autoware_auto_vehicle_msgs/msg/Engage "engage: true" -1
+```
+
+![start-driving](images/planning/lane-following/engage-and-start-planning.png)
+
+### Parking scenario
 
 1. Set an initial pose and a goal pose, and engage the ego vehicle.
 
    ![after-set-goal-pose](images/planning/parking/after-set-goal-pose.png)
 
 2. When the vehicle approaches the goal, it will change to the parking mode.
+3. After that, the vehicle will start driving backwards and park at the destination parking spot.
 
-   ![change-to-parking-mode](images/planning/parking/change-to-parking-mode.png)
+   ![parking-maneuver](images/planning/parking/parking-maneuver.png)
 
-3. After that, the vehicle will start driving backwards and park at the destination parking spot.
+## Advanced Simulations
+
+### Placing dummy objects
+
+1. Click the `2D Dummy Car` or `2D Dummy Pedestrian` button in the toolbar.
+
+2. Set the pose of the dummy object by clicking and dragging on the map.
+
+3. Set the velocity of the object in `Tool Properties -> 2D Dummy Car/Pedestrian` panel.
+
+![set-dummy-car](images/planning/lane-following/place-dummy-car.png) 4. Delete any dummy objects placed in the view by clicking the `Delete All Objects` button in the toolbar.
+
+### Traffic light recognition simulation
+
+#### Set traffic light
+
+1. Go to `Panels -> Add new panel`, select `TrafficLightPublishPanel`, and then press `OK`.
+
+2. In `TrafficLightPublishPanel`, set the `ID` and color of the traffic light.
+
+3. Click the `SET` button.
+
+![set-traffic-light](images/planning/lane-following/set-traffic-light.png) 4. Finally, click the `PUBLISH` button to send the traffic light status to the simulator. Any planned path that goes past the selected traffic light will then change accordingly.
```

Comment:
```suggestion
![set-traffic-light](images/planning/lane-following/set-traffic-light.png) 

4. Finally, click the `PUBLISH` button to send the traffic light status to the simulator. Any planned path that goes past the selected traffic light will then change accordingly.
```

Reply:
I had a check, and it looks like the change happened here:
![image](https://user-images.githubusercontent.com/63835446/173083879-fecfcd10-1a96-4787-bfb6-7c5ddd3a51e0.png)

https://github.com/autowarefoundation/autoware-documentation/compare/1125f66f8cf18dad9907307f2e2ab85e678b09e1..d376f5d3739dbe600acd10b098c7325dde81ff25